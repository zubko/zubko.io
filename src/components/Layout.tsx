/**
 * Root component of the page
 */

import { graphql, StaticQuery } from 'gatsby';
import { ReactNode } from 'react';

import { Footer, Header } from '../components';
import { rhythm } from '../Typography';

type Props = {
  children: ReactNode;
};

const Layout = ({ children }: Props) => (
  <StaticQuery
    query={graphql`
      query SiteTitleQuery {
        site {
          siteMetadata {
            title
          }
        }
      }
    `}
    render={data => {
      return (
        <div
          css={{
            margin: `0 auto`,
            maxWidth: rhythm(24),
            padding: `0px 1.0875rem 1.45rem`,
            paddingTop: 0,
          }}>
          <header>
            <Header siteTitle={data.site.siteMetadata.title} />
          </header>
          <main>{children}</main>
          <Footer />
        </div>
      );
    }}
  />
);

export default Layout;
