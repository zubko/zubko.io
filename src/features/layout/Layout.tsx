/**
 * Root component of the page
 */

import { graphql, StaticQuery } from "gatsby";
import { ReactNode } from "react";
import styled from "styled-components";

import { rhythm } from "../../Typography";
import { Footer } from "./Footer";
import { Header } from "./Header";

type Props = {
  children: ReactNode;
};

export const Layout = ({ children }: Props) => (
  <StaticQuery
    query={graphql`
      query SiteTitleQuery {
        site {
          siteMetadata {
            title
          }
        }
      }
    `}
    render={(data) => {
      return (
        <Container>
          <header>
            <Header siteTitle={data.site.siteMetadata.title} />
          </header>
          <main>{children}</main>
          <Footer />
        </Container>
      );
    }}
  />
);

const Container = styled.div`
  margin: 0 auto;
  max-width: ${rhythm(24)};
  padding: 0px 1.0875rem 1.45rem;
  padding-top: 0px;
`;
