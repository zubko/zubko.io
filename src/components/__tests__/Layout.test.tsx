import React from "react";
import renderer from "react-test-renderer";
import { StaticQuery } from "gatsby";

import Layout from "../Layout";

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render({
      site: {
        siteMetadata: {
          title: "title",
        },
      },
    })
  ).mockImplementationOnce(({ render }) =>
    render({
      allMenuYaml: {
        edges: [
          {
            node: {
              path: "/path-1/",
              title: "title-1",
            },
          },
          {
            node: {
              path: "/path-2/",
              title: "title-2",
            },
          },
        ],
      },
    })
  );
});

describe("Layout", () => {
  it("renders without regressions", () => {
    const tree = renderer.create(<Layout>Child</Layout>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
