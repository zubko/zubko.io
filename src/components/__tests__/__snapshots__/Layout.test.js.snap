// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders without regressions 1`] = `
<div
  className="css-3yvbqg-Layout"
>
  <header>
    <div
      className="css-1pcmc93"
    >
      <div
        className="css-13o4jer"
      >
        <a
          className="css-9bb8dy-linkPlain"
          href="/"
        >
          <h3>
            title
          </h3>
        </a>
        <div
          className="css-albw4v-MenuHorizontal-Header"
        >
          <a
            className="css-kbxh37-linkPlain"
            href="/path-1/"
          >
            title-1
          </a>
          <span
            className="css-x4n1ct-Separator"
          >
             
          </span>
          <a
            className="css-kbxh37-linkPlain"
            href="/path-2/"
          >
            title-2
          </a>
        </div>
        <div
          className="css-1hw5nl3-Header"
        >
          <button
            className="css-91zqk4-MenuVertical"
            onClick={[Function]}
          >
            Menu
          </button>
        </div>
      </div>
      <div
        className="css-1hw5nl3-Header"
      >
        <div
          className="css-1h6xv3r-MenuVertical"
        >
          <a
            className="css-kbxh37-linkPlain"
            href="/path-1/"
          >
            title-1
          </a>
          <span
            className="css-1gnmjk0-Separator"
          >
             
          </span>
          <a
            className="css-kbxh37-linkPlain"
            href="/path-2/"
          >
            title-2
          </a>
        </div>
      </div>
    </div>
  </header>
  <main>
    Child
  </main>
  <footer
    className="css-1o3tyej"
  >
    <p>
      ~~~
      <br />
       © 
      2019
      , Built
       
      <a
        className="css-1a4dihx-linkPlain"
        href="/acknowledgements"
      >
        with ❤️and open source ⚛
      </a>
    </p>
  </footer>
</div>
`;
